src/CloudEvents/V1/CloudEvent.php

```diff
@@ -0,0 +1,150 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CloudEvents\V1;
+
+use DateTimeInterface;
+
+class CloudEvent implements CloudEventInterface
+{
+    private const SPEC_VERSION = '1.0';
+
+    private string $id;
+    private string $source;
+    private string $type;
+    private ?string $dataContentType;
+    private ?string $dataSchema;
+    private ?string $subject;
+    private ?DateTimeInterface $time;
+
+    /** @var mixed|null */
+    private $data;
+
+    public function __construct(
```

Comment:
I thought about having just the required properties be in the constructor, but thinking forward to PHP 8, I feel this constructor signature will play nicely with named parameters and property promotion.